### PR TITLE
feat(dummy_perception_publisher): publish ground truth object in option

### PR DESCRIPTION
**Pull request by YoshiRi**
## Description

Dummy detections are generated by removing undetected parts from the ground truth with a certain probability, and then adding appropriate noise.

- This PR added option to
  - Publish ground truth as a TrackedObjects
  - Enable to fix the random seed used in noise generation.

## Tests performed

Tested in Psim.

- pink bounding box: ground truth
- blue bounding box: dummy detection
image

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: simulator/dummy_perception_publisher/README.md [hunk 1]
```diff
@@ -17,10 +17,11 @@ This node publishes the result of the dummy detection with the type of perceptio
...
 
-| Name                    | Type                                                     | Description            |
-| ----------------------- | -------------------------------------------------------- | ---------------------- |
-| `output/dynamic_object` | `tier4_perception_msgs::msg::DetectedObjectsWithFeature` | Publishes objects      |
-| `output/points_raw`     | `sensor_msgs::msg::PointCloud2`                          | point cloud of objects |
+| Name                          | Type                                                     | Description                     |
+| ----------------------------- | -------------------------------------------------------- | ------------------------------- |
+| `output/dynamic_object`       | `tier4_perception_msgs::msg::DetectedObjectsWithFeature` | dummy detection objects         |
+| `output/points_raw`           | `sensor_msgs::msg::PointCloud2`                          | point cloud of objects          |
+| `output/ground_truth_objects` | `autoware_auto_perception_msgs::msg::TrackedObjects`     | ground truth objects (optional) |
```
Comment (yukkysaito) [inline simulator/dummy_perception_publisher/README.md:24]:
```suggestion
| `output/debug/ground_truth_objects` | `autoware_auto_perception_msgs::msg::TrackedObjects`     | ground truth objects |
```
How about this?

File: simulator/dummy_perception_publisher/src/node.cpp [hunk 2]
```diff
@@ -134,6 +166,13 @@ DummyPerceptionPublisherNode::DummyPerceptionPublisherNode()
     "input/object", 100,
     std::bind(&DummyPerceptionPublisherNode::objectCallback, this, std::placeholders::_1));
 
+  // optional ground truth publisher
+  if (publish_ground_truth_objects_) {
+    ground_truth_objects_pub_ =
+      this->create_publisher<autoware_auto_perception_msgs::msg::TrackedObjects>(
+        "output/ground_truth_objects", qos);
+  }
+
```
Comment (yukkysaito) [inline simulator/dummy_perception_publisher/src/node.cpp:175]:
```suggestion
  ground_truth_objects_pub_ =
    this->create_publisher<autoware_auto_perception_msgs::msg::TrackedObjects>(
        "~output/debug/ground_truth_objects", qos);

```

Comment (YoshiRi) [reply simulator/dummy_perception_publisher/src/node.cpp]: Also done in https://github.com/autowarefoundation/autoware.universe/pull/3731/commits/a03800be2915f12c08e48b5b86b9abbdf093941b.

**Review by yukkysaito — APPROVED** [review]
LGTM

**Pull request merged**